### PR TITLE
[RTL] More Module cases

### PR DESCRIPTION
**Pull request by darthscsi**
Support emission of modules when not wrapped in a firrtl.circuit (e.g. firrtl-free rtl dialect).
Fix missing ';' in instance lowering.

### CODE REVIEW

**Review by lattner — APPROVED** [review]

File: lib/EmitVerilog/EmitVerilog.cpp [hunk 1]
```diff
@@ -1629,24 +1631,39 @@ void ModuleEmitter::emitStatement(rtl::RTLInstanceOp op) {
...
+  if (auto moduleIR = op.getParentOfType<firrtl::CircuitOp>()) {
+    auto moduleOp = moduleIR.lookupSymbol(defName);
+    if (auto m = cast<rtl::RTLModuleOp>(moduleOp)) {
+      m.getRTLPortInfo(portInfo);
+    } else if (auto m = cast<rtl::RTLExternModuleOp>(moduleOp)) {
+      m.getRTLPortInfo(portInfo);
+    } else {
+      assert(0 && "invalid rtl.instance op");
+    }
+  } else if (auto moduleIR = op.getParentOfType<mlir::ModuleOp>()) {
```
Comment (lattner) [inline lib/EmitVerilog/EmitVerilog.cpp:1644]: Actually, I don't think we want to support free standing modules.  All firrtl.module declaration should be in a circuit - we should add verifier check for that.

Comment (lattner) [reply lib/EmitVerilog/EmitVerilog.cpp:1644]: Oh wait, I got it - this looks right.  `rtl.module` shouldn't be required to be in a `firrtl.circuit`.  In fact, maybe they shouldn't be *allowed* to be.  :-)

Comment (darthscsi) [reply lib/EmitVerilog/EmitVerilog.cpp:1644]: firrtl moduels should be in a circuit, but rtl modules don't need to be.

Comment (darthscsi) [reply lib/EmitVerilog/EmitVerilog.cpp:1644]: Which is what is done now.
firrtl.circuit { rtl.module, rtl.externmodule, firrtl.module}
mlir.module { rtl.module, rtl.externmodule}

Comment (teqdruid) [reply lib/EmitVerilog/EmitVerilog.cpp:1668]: Would be better if this was enforced at IR parsing / verification time.

Comment (darthscsi) [reply lib/EmitVerilog/EmitVerilog.cpp:1668]: Yes, this patch simply increases the set of options not rejected at print time to include the things we want.  Eventually this pass won't have any firrtl support.

**Pull request merged**